EIPS/eip-5007.md

```diff
@@ -0,0 +1,313 @@
+---
+eip: 5007
+title: TimeNFT, ERC-721 Time Extension
+description: Add start time and end time to ERC-721 tokens.
+author: Anders (@0xanders), Lance (@LanceSnow), Shrug <shrug@emojidao.org>
+discussions-to: https://ethereum-magicians.org/t/eip5007-erc721-time-extension/8924
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-04-13
+requires: 165, 721
+---
+
+## Abstract
+
+This standard is an extension of [ERC-721](./eip-721.md). It proposes some additional property( `startTime`, `endTime`,`originalTokenId`) to help with the on-chain time management.
+
+## Motivation
+
+Some NFTs have a defined usage period and cannot be used when they are not at a specific time. If you want to make NFT invalid when it is not in use period, or make NFT enabled at a specific time, while the NFT does not contain time information, you often need to actively submit the chain transaction, this process is both cumbersome and a waste of gas.
+
+There are also some NFTs contain time functions, but the naming is different, third-party platforms are difficult to develop based on it.
+
+By introducing (`startTime`, `endTime`) and unifying the naming, it is possible to enable and disable NFT automatically on chain.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+interface IERC5007  {
+
+    /// @notice Emitted when the `startTime` or `endTime` of a NFT is changed
+    /// @param tokenId  The tokenId of the NFT
+    /// @param startTime  The new start time of the NFT
+    /// @param endTime  The new end time of the NFT
+    event TimeUpdate(uint256 tokenId, uint64 startTime, uint64 endTime);
+
+    /// @notice Get the start time of the NFT
+    /// @dev Throws if `tokenId` is not valid NFT
+    /// @param tokenId  The tokenId of the NFT
+    /// @return The start time of the NFT
+    function startTime(uint256 tokenId) external view returns (uint64);
+    
+    /// @notice Get the end time of the NFT
+    /// @dev Throws if `tokenId` is not valid NFT
+    /// @param tokenId  The tokenId of the NFT
+    /// @return The end time of the NFT
+    function endTime(uint256 tokenId) external view returns (uint64);
```

Comment:
If `block.timestamp == IERC5007.endTime(0x...)`, is the token valid or invalid?

Reply:
Also note this too. I know it's in the comment for `isValidNow`, but it could be more visible.